packages/SwingSet/src/controller.js

```diff
@@ -37,6 +43,51 @@ function makeConsole(tag) {
   return harden(cons);
 }
 
+function makeStartXSnap(bundles, { env }) {
+  const xsnapOpts = {
+    os: osType(),
+    spawn,
+    stdout: 'inherit',
+    stderr: 'inherit',
+    debug: !!env.XSNAP_DEBUG,
+  };
+
+  const pool = defaultSnapstorePath({ env });
+  fs.mkdirSync(pool, { recursive: true });
+
+  const snapStore = makeSnapstore(pool, {
+    tmpName,
+    existsSync: fs.existsSync,
+    createReadStream: fs.createReadStream,
+    createWriteStream: fs.createWriteStream,
+    rename: fs.promises.rename,
+    unlink: fs.promises.unlink,
+    resolve: path.resolve,
+  });
+
+  let supervisorHash = '';
+  return async function startXSnap(name, handleCommand) {
+    if (supervisorHash) {
+      return snapStore.load(supervisorHash, async snapshot => {
```

Comment:
Ah, right, you have to pass in a function (rather than making `snapStore.load` return a Promise) because you need a place to delete the tempfile afterwards. That's a pity. If/when we decide to pass snapshots as data over a pipe (to enable `seccomp(2)` #2386), and we have to hold the snapshot in RAM otherwise, let's consider changing this API to be a simple return-a-Promise one.